includes/MediaWiki/Request/EditRequestParser.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request;
+
+use MediaWiki\Extension\WikibaseReconcileEdit\InputToEntity\FullWikibaseItemInput;
+use MediaWiki\Extension\WikibaseReconcileEdit\InputToEntity\MinimalItemInput;
+use MediaWiki\Rest\LocalizedHttpException;
+use MediaWiki\Rest\RequestInterface;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\Lookup\PropertyDataTypeLookup;
+use Wikimedia\Message\MessageValue;
+
+/**
+ * @license GPL-2.0-or-later
+ */
+class EditRequestParser {
+
+	public const VERSION_KEY = "wikibasereconcileedit-version";
+
+	private const SUPPORTED_ENTITY_VERSIONS = [ '0.0.1/full', '0.0.1/minimal' ];
+	private const SUPPORTED_RECONCILE_VERSIONS = [ '0.0.1' ];
+
+	/** @var PropertyDataTypeLookup */
+	private $propertyDataTypeLookup;
+
+	/** @var FullWikibaseItemInput */
+	private $fullWikibaseItemInput;
+
+	/** @var MinimalItemInput */
+	private $minimalItemInput;
+
+	public function __construct(
+		PropertyDataTypeLookup $propertyDataTypeLookup,
+		FullWikibaseItemInput $fullWikibaseItemInput,
+		MinimalItemInput $minimalItemInput
+	) {
+		$this->propertyDataTypeLookup = $propertyDataTypeLookup;
+		$this->fullWikibaseItemInput = $fullWikibaseItemInput;
+		$this->minimalItemInput = $minimalItemInput;
+	}
+
+	public function parseRequestInterface( RequestInterface $request ): EditRequest {
+		$entity = json_decode(
+			$request->getPostParams()['entity'],
+			true
+		);
+
+		if (
+			// TODO separate error message for invalid entity JSON?
+			!is_array( $entity ) ||
+			!array_key_exists( self::VERSION_KEY, $entity )
+		) {
+			throw new LocalizedHttpException(
```

Comment:
Feels a bit weird that this thing throws LocalizedHttpException with a status code (which seems to never change).

Could see this throwing some ParseError that the API endpoint then decides the status code for. What do you think?

Reply:
Then again it's parsing requests so maybe it makes sense for now.

Reply:
Yeah, I concluded in [T282564#7087599](https://phabricator.wikimedia.org/T282564#7087599) that there wasn’t a great alternative solution and that an HTTP exception was hopefully acceptable in a request parser.

Reply:
Cool, lets leave it for now then.